src/ucp/wireup/wireup_ep.c

```diff
@@ -331,33 +331,60 @@ static ucs_status_t ucp_wireup_ep_flush(uct_ep_h uct_ep, unsigned flags,
     return UCS_ERR_NO_RESOURCE;
 }
 
+static ucs_status_t
+ucp_wireup_ep_do_check(ucp_ep_h ucp_ep, uct_ep_h checked_uct_ep,
+                       uct_ep_h uct_ep, ucp_rsc_index_t rsc_idx, unsigned flags,
+                       uct_completion_t *comp, int strict_ep_check)
+{
+    ucp_worker_h worker = ucp_ep->worker;
+    ucp_worker_iface_t *wiface;
+    ucp_rsc_index_t checked_rsc_idx;
+    ucp_lane_index_t checked_lane;
+
+    ucs_assert(rsc_idx != UCP_NULL_RESOURCE);
+
+    checked_lane = ucp_ep_lookup_lane(ucp_ep, checked_uct_ep);
+    ucs_assert(checked_uct_ep == ucp_ep->uct_eps[checked_lane]);
+
+    checked_rsc_idx = ucp_ep_get_rsc_index(ucp_ep, checked_lane);
+
+    wiface = ucp_worker_iface(worker, rsc_idx);
+    ucs_assert(!strict_ep_check ||
+               (wiface->attr.cap.flags & UCT_IFACE_FLAG_EP_CHECK));
+
+    if (wiface->attr.cap.flags & UCT_IFACE_FLAG_EP_CHECK) {
+        if ((ucp_ep->flags & UCP_EP_FLAG_REMOTE_ID) &&
+            (wiface->attr.cap.flags & UCT_IFACE_FLAG_CONNECT_TO_IFACE)) {
+            return ucp_wireup_msg_send(ucp_ep, UCP_WIREUP_MSG_EP_CHECK,
+                                       UCS_BIT(checked_rsc_idx), NULL);
+        }
+        return uct_ep_check(uct_ep, flags, comp);
+    }
+
+    /* if EP_CHECK is not supported by UCT transport, it has to support a
+     * built-in keepalive mechanism to be able to detect peer failure during
+     * wireup
+     */
+    ucs_assert(wiface->attr.cap.flags & UCT_IFACE_FLAG_EP_KEEPALIVE);
+    return UCS_OK;
```

Comment:
can we unite common code with ucp ep keepalive?

Reply:
will think how to separate the common part and use it here and there